crates/oxc_linter/src/tester.rs

```diff
@@ -165,6 +167,31 @@ where
     }
 }
 
+struct TesterFileSystem {
+    path_to_lint: PathBuf,
+    source_text: String,
+}
+
+impl TesterFileSystem {
+    pub fn new(path_to_lint: PathBuf, source_text: String) -> Self {
+        Self { path_to_lint, source_text }
+    }
+}
+
+impl RuntimeFileSystem for TesterFileSystem {
+    fn read_to_string(&self, path: &Path) -> Result<String, std::io::Error> {
+        if path == self.path_to_lint {
+            return Ok(self.source_text.clone());
+        }
+
+        read_to_string(path)
+    }
+
+    fn write_file(&self, _path: &Path, _content: String) -> Result<(), std::io::Error> {
+        unreachable!("writing file should not be allowed in Tester");
```

Comment:
nit: unreachable is probably not correct here - i think just a `panic!()` is better here